qulice-maven-plugin/src/test/java/com/qulice/maven/DefaultMavenEnvironmentTest.java

```diff
@@ -128,4 +128,30 @@ public final void passPathsWithWhitespaces()  throws Exception {
             Matchers.notNullValue()
         );
     }
+
+    /**
+     * DefaultMavenEnvironment can produce empty collection when no matches
+     * with checker.
+     * @throws Exception If something wrong happens inside
+     * @todo Ideally integration test required to reproduce NPE described in
+     *  #676. It should contains at least one exclude rule in
```

Comment:
@vkuchyn please also state that in order to reproduce bug, changes in `DefaultMavenEnvironment.java` must be reverted